### PR TITLE
DMP-1384 - Cannot approve or reject your own transcript requests.

### PR DESCRIPTION
**Pull request by HMCTSDeveloper**
https://tools.hmcts.net/jira/browse/DMP-1384

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/transcriptions/exception/TranscriptionApiError.java
Comment (johnrace0) [inline src/main/java/uk/gov/hmcts/darts/transcriptions/exception/TranscriptionApiError.java:1]: maybe rename REQUESTER_CANNOT_BE_APPROVER_OR_REJECTER

File: src/main/java/uk/gov/hmcts/darts/transcriptions/service/impl/TranscriptionServiceImpl.java [hunk 1]
```diff
@@ -260,6 +260,10 @@ private void handleNotificationsAndAudit(UserAccountEntity userAccountEntity,
     private void validateUpdateTranscription(TranscriptionEntity transcription,
                                              UpdateTranscription updateTranscription) {
 
+
+        if (getUserAccount().getUserName().equals(transcription.getRequestor())) {
```
Comment (johnrace0) [inline src/main/java/uk/gov/hmcts/darts/transcriptions/service/impl/TranscriptionServiceImpl.java:264]: is it definitely .equals or .equalsIgnoreCase - this tripped me up on a bug recently

File: src/main/java/uk/gov/hmcts/darts/transcriptions/service/impl/TranscriptionServiceImpl.java [hunk 2]
```diff
@@ -634,9 +638,8 @@ private List<TranscriptionEntity> processTranscriptionUpdates(List<UpdateTranscr
         }
 
         if (!validated.isEmpty()) {
-            validated.stream().forEach(entity -> {
-                UpdateTranscriptionEntityHelper.updateTranscriptionEntity(entity, getTranscriptionsItemForId(entity.getId(), request).get());
-            });
+            validated.forEach(entity -> UpdateTranscriptionEntityHelper.updateTranscriptionEntity(entity, getTranscriptionsItemForId(entity.getId(), request)
+                                      .get()));
```
Comment (johnrace0) [inline src/main/java/uk/gov/hmcts/darts/transcriptions/service/impl/TranscriptionServiceImpl.java:642]: formatting looks a bit weird here

File: src/integrationTest/java/uk/gov/hmcts/darts/transcriptions/controller/TranscriptionControllerUpdateTranscriptionApprovedIntTest.java [hunk 3]
```diff
@@ -241,4 +241,32 @@ void updateTranscriptionShouldReturnTranscriptionWorkflowActionInvalidError() th
         verifyNoInteractions(mockAuditApi);
     }
 
+
+    @Test
+    void updateTranscriptionApprovedWithRequestorSameAsApprover() throws Exception {
```
Comment (johnrace0) [inline src/integrationTest/java/uk/gov/hmcts/darts/transcriptions/controller/TranscriptionControllerUpdateTranscriptionApprovedIntTest.java:246]: maybe give the new test naming conventions a try?

File: src/main/java/uk/gov/hmcts/darts/transcriptions/service/impl/TranscriptionServiceImpl.java (hunk 2 above)
Comment (HMCTSDeveloper) [reply src/main/java/uk/gov/hmcts/darts/transcriptions/service/impl/TranscriptionServiceImpl.java:642]: ACCEPT.

File: src/integrationTest/java/uk/gov/hmcts/darts/transcriptions/controller/TranscriptionControllerUpdateTranscriptionApprovedIntTest.java (hunk 3 above)
Comment (HMCTSDeveloper) [reply src/integrationTest/java/uk/gov/hmcts/darts/transcriptions/controller/TranscriptionControllerUpdateTranscriptionApprovedIntTest.java]: ACCEPT.

File: src/main/java/uk/gov/hmcts/darts/transcriptions/exception/TranscriptionApiError.java
Comment (HMCTSDeveloper) [reply src/main/java/uk/gov/hmcts/darts/transcriptions/exception/TranscriptionApiError.java:1]: ACCEPT

**Review by johnrace0 — APPROVED** [review]

**Pull request merged**